pkg/maps/vtep/vtep.go

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package vtep
+
+import (
+	"fmt"
+	"net"
+	"unsafe"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/cilium/cilium/pkg/cidr"
+	"github.com/cilium/cilium/pkg/defaults"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/mac"
+	"github.com/cilium/cilium/pkg/types"
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "map-vtep")
+
+const (
+	// MaxEntries is the maximum number of keys that can be present in the
+	// VTEP map.
+	MaxEntries = defaults.MaxVTEPDevices
+
+	// Name is the canonical name for the VTEP map on the filesystem.
+	Name = "cilium_vtep_map"
+)
+
+// Key implements the bpf.MapKey interface.
+//
+// Must be in sync with struct vtep_key in <bpf/lib/common.h>
+// +k8s:deepcopy-gen=true
+// +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapKey
+type Key struct {
+	IP types.IPv4 `align:"vtep_ip"`
+}
+
+// GetKeyPtr returns the unsafe pointer to the BPF key
+func (k *Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+// NewValue returns a new empty instance of the structure representing the BPF
+// map value
+func (k Key) NewValue() bpf.MapValue { return &VtepEndpointInfo{} }
+
+func (k Key) String() string {
+	ipStr := net.IP(k.IP[:net.IPv4len]).String()
+	return fmt.Sprintf("%s", ipStr)
```

Comment:
This can be simplified:

```suggestion
	return k.IP.String()
```